libkbfs/folder_block_ops.go

```diff
@@ -1315,8 +1315,8 @@ func (fbo *folderBlockOps) writeDataLocked(
 			de.EncodedSize = 0
 			// update the file info
 			de.Size += uint64(len(block.Contents) - oldLen)
-			fbo.deCache[file.tailPointer().ref()] = de
 		}
+		fbo.deCache[file.tailPointer().ref()] = de
```

Comment:
Can you add a comment to `deCache` in its definition in `folderBlockOps` mentioning that it's used to determine the overall clean vs. dirty state? It's a bit surprising, since now we have the `dirtyFiles` field, too.

Speaking of which, (not for this PR) should we eventually combine `dirtyFiles`, `unrefCache`, and `deCache`, i.e. have `dirtyFile` also hold a `DirEntry` and a `syncInfo`?

Also not for this PR, but is there any reason `dirtyFiles` has a `BlockPointer` key instead of a `blockRef` key? Does it need to distinguish fields other than id and refNonce?